core/lib/rom/attribute.rb

```diff
@@ -24,6 +24,10 @@ class Attribute
     #   @return [Dry::Types::Definition, Dry::Types::Sum, Dry::Types::Constrained] The attribute's type object
     param :type
 
+    # @!attribute [r] type
+    #   @return [Symbol] Attribute name. This is the canonical name used in the datastore.
+    option :name, type: Types::Strict::Symbol
+
```

Comment:
Being an option, `#name` method is no longer needed.

Reply:
Ok, I thought that maybe it was too much for an attribute. But you are right, in fact it makes no difference whether it is an attribute or a method. However, now that I give it a second view, I think the example given is incorrect:

```ruby
    # @example	
    #   class Tasks < ROM::Relation[:memory]	
    #     schema do	
    #       attribute :user_id, Types::Integer.meta(alias: :id)	
    #       attribute :name, Types::String	
    #     end	
    #   end	
    #	
    #   Users.schema[:id].name	
    #   # => :id	
    #	
    #   Users.schema[:user_id].name	
    #   # => :user_id
```

`:id` attribute is not defined in the schema, and `:user_id` name should return `:id`, isn't it? Maybe  should we just change it to:

```ruby
  # Users.schema[:name].name
  # # => :name
  #
  # Users.schema[:id].name
  # # => :id
```

?

Reply:
@waiting-for-dev hah good catch, let's just stick to this:

```
    # @example	
    #   class Users < ROM::Relation[:memory]	
    #     schema(:users) do	
    #       attribute :user_id, Types::Integer, alias: :id
    #       attribute :email, Types::String
    #     end	
    #   end	
    #	
    #   users[:id].name	
    #   # => :id
    #
    #   users[:email].name	
    #   # => :email
```